### ISSUE TITLE
[css-mediaqueries-4] should color-gamut be evaluated in a boolean context?

### ISSUE THREAD

**Issue by tbjgolden** [w3c/csswg-drafts#7121]
At https://www.w3.org/TR/mediaqueries-4/#mq-boolean-context the spec says

> media features [...] can also be written more simply as just the feature name, like `(color)`.
>
> When written like this, the [media feature](https://www.w3.org/TR/mediaqueries-4/#media-feature) is evaluated in a boolean context.
>
> If the feature would be true for any value other than the number 0, a [<dimension>](https://www.w3.org/TR/css-values-3/#typedef-dimension) with the value 0, or the keyword [none](https://www.w3.org/TR/mediaqueries-4/#valdef-media-update-none), the media feature evaluates to true.

If this is the case, then why does it say (in https://www.w3.org/TR/mediaqueries-4/#color-gamut):

> Note: Some output devices, such as monochrome displays, cannot support even the [srgb](https://www.w3.org/TR/mediaqueries-4/#valdef-media-color-gamut-srgb) gamut. To test for these devices, you can use this feature in a negated boolean-context fashion: `not (color-gamut)`.

---

This means that either:

These two sections contradict each other - which one is true? If it's the `not (color-gamut)` should the boolean context section be changed?

**Comment by tabatkins** [comment]
The boolean-feature condition is *not* that the feature evaluates to true for one of the "falsey" values, but that it *not* evaluate to true for any of the *truthy* values. If it lacks any falsey values entirely (as in `color-gamut`), it can be false in a boolean context so long as none of its truthy values match.

(That said, this is the sort of clever "technically correct" thing that specs shouldn't depend on; I'll make this case clearer in the spec.)

**Comment by tbjgolden** [comment]
it might also be useful to point out that `(grid)` is equivalent to `(grid: 0)` :)

**Comment by tabatkins** [comment]
What do you mean? It would be equivalent to `(grid: 1)`. Why do you think this is an exception?

**Comment by tbjgolden** [comment]
Referring to the Note below Example 32:

https://www.w3.org/TR/mediaqueries-4/#example-07e10c7e

The spec says it's supposed to be `(grid: 1)` but every browser treats `(grid)` as `(grid: 0)`

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/grid

**Comment by tbjgolden** [comment]
No problem! :)

Chrome does support it (but because your display is not a "grid" display like a terminal) it's not truthy

Chrome should work with `@media not (grid)` though - which would indicate it does support it

**Comment by tabatkins** [comment]
That appears to be correct - that's not a grid browser, so `(grid: 0)` is true and `(grid: 1)` is false, so `not (grid)` is true, and it is correctly matching in your example.

**Comment by tbjgolden** [comment]
So after checking a bit more

### Firefox

- `not (grid)` matches
- `not (grid: 0)` does not match
- `not (grid: 1)` matches

### Chrome

- `(grid: 0)` matches
- `(grid)` does not match
- `(grid: 1)` does not match
- `not (grid: 1)` does not match

**Comment by tabatkins** [comment]
Right, so aside from Chrome not matching `(grid)` in a boolean context at all (thus treating it as an unknown and thus false feature, I presume), all of this is exactly as it should be; I'm not seeing any exceptions to the existing rule.

**Comment by tbjgolden** [comment]
> I'm not seeing any exceptions to the existing rule

Yup, no exceptions to the existing rule

It looks as if Chrome has bugs with `not` unrelated to their grid logic

---

Conclusions (correct me if I'm wrong)

- the Note below Example 32 (https://www.w3.org/TR/mediaqueries-4/#example-07e10c7e) should be removed as it isn't true
- the mdn data is incomplete and should probably also check for boolean support - I can try and get that fixed with them

**Comment by tabatkins** [comment]
No, that note is correct - that MQ will indeed detect precisely what the text says it will. (I just don't think there's any device running a browser that executes MQs which will ever return true for it.)

**Comment by tbjgolden** [comment]
Oh! I understand. :man_facepalming: 

I was thinking that the note was trying to tell me that `(grid)` is treated as `(grid: 0)`!

> Note: At the time of writing, all known implementations match grid: 0 rather than grid: 1.

I read implementations as "implementations of boolean (grid)" rather than "implementations of media queries generally", probably because above it boolean grid is shown in an example and I was subconsciously expecting to read something about boolean grid.

Will let you decide if rephrasing is worth it or not, but in any case thanks for helping me understand what I was missing!
